rdb/src/main/java/org/openmbee/mms/rdb/repositories/commit/CommitDAOImpl.java

```diff
@@ -156,7 +157,7 @@ public List<Commit> findByRefAndTimestampAndLimit(Branch ref, Instant timestamp,
             currentRef = ref.getParentRefId();
             currentCid = ref.getParentCommit();
 
-            if (currentRef == null) {
+            if (currentRef == null || currentRef.equals(Constants.MASTER_BRANCH)) {
```

Comment:
This breaks the tests since it stops at a branch whose parent is master.
master branch is created with parent ref being null, it may be you had some bad db data from a dev branch for some reason had master with master parent